lib/api3/generic/search/operation.js

```diff
@@ -43,11 +42,9 @@ async function search (opCtx) {
       , onlyValid });
 
     if (!result)
-      throw new Error('empty result');
+      throw new Error('empty result');    result.forEach(col.resolveDates);
```

Comment:
Accidental add? The forEach doesn't do anything

Reply:
No, the `resolveDates` method is defined in lib/api3/generic/collection.js and needs to be called on each of the results.